stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/header/MessageListHeader.kt

```diff
@@ -86,55 +98,58 @@ public fun MessageListHeader(
             verticalAlignment = Alignment.CenterVertically,
         ) {
 
-            BackButton(
-                modifier = Modifier
-                    .size(36.dp),
-                imageVector = Icons.Default.ArrowBack,
-                onBackPressed = onBackPressed
-            )
+            leadingContent()
 
-            MessagesHeaderTitle(
-                modifier = Modifier
-                    .weight(6f)
-                    .fillMaxHeight()
-                    .clickable(
-                        interactionSource = remember { MutableInteractionSource() },
-                        indication = null,
-                        onClick = onHeaderActionClick
-                    ),
-                title = title,
-                subtitle = subtitle,
-                isNetworkAvailable = isNetworkAvailable
-            )
+            titleContent()
 
-            ChannelAvatar(
-                modifier = Modifier.size(36.dp),
-                channel = channel,
-                currentUser = currentUser,
-                contentDescription = channel.name
-            )
+            trailingContent()
         }
     }
 }
 
 /**
- * Wrapper for the title of the header, that handles if we should show a loading view for network,
+ * Default header title, that handles if we should show a loading view for network,
  * or the channel information.
  *
+ * @param channel The channel used for the title information.
  * @param modifier Modifier for styling.
- * @param title The title of the header.
- * @param subtitle The subtitle of the header - usually information about channel members.
+ * @param messageMode Currently active message mode, used to define the title information.
+ * @param onHeaderActionClick Handler for when the user taps on the header content.
  * @param isNetworkAvailable A flag that governs if we show the subtitle or the network loading view.
  */
 @Composable
-private fun MessagesHeaderTitle(
+public fun DefaultMessageHeaderTitle(
+    channel: Channel,
     modifier: Modifier,
-    title: String,
-    subtitle: String,
-    isNetworkAvailable: Boolean,
+    messageMode: MessageMode = Normal,
+    onHeaderActionClick: (Channel) -> Unit = {},
+    isNetworkAvailable: Boolean = true,
 ) {
+    val memberCount = channel.memberCount
+    val onlineCount = channel.members.count { it.user.online }
+
+    val channelName = channel.getDisplayName()
+
+    val title = if (messageMode == Normal) {
+        channelName
+    } else {
+        stringResource(id = R.string.stream_compose_thread_title)
+    }
+
+    val subtitle = if (messageMode == Normal) {
+        stringResource(id = R.string.stream_compose_channel_members, memberCount, onlineCount)
+    } else {
+        stringResource(id = R.string.stream_compose_thread_subtitle, channelName)
+    }
```

Comment:
We can make the `else` case more obvious with `when`s here

```suggestion
    val title = when (messageMode) {
        Normal -> channelName
        is Thread -> stringResource(id = R.string.stream_compose_thread_title)
    }

    val subtitle = when (messageMode) {
        Normal -> stringResource(id = R.string.stream_compose_channel_members, memberCount, onlineCount)
        is Thread -> stringResource(id = R.string.stream_compose_thread_subtitle, channelName)
    }
```

Reply:
The suggestion here was specifically to have `is Thread` as the condition on the second branches. Otherwise, it's "what to do in normal mode and what to do otherwise", but it might not be immediately clear what the "otherwise" case means if you don't know this code very well.

Reply:
Ohh okay that makes more sense now. I'll update it accordingly!